Nicknames contain the value null

Nico8340 opened the pull request:
Nicknames sometimes contain the word null, because there are fewer entries in the arrays at the specified length. Fixes #3832.
Nouns and adjectives have been counted with this piece of code:
```js
const adjectives = [
    "Aback",      "Abaft",      "Abandoned",  "Abashed",    "Aberrant",   "Abhorrent",  "Abiding",    "Abject",     "Ablaze",     "Able",       "Abnormal",
    "Aboard",     "Aboriginal", "Abortive",   "Abounding",  "Abrasive",   "Abrupt",     "Absent",     "Absorbed",   "Absorbing",  "Abstracted", "Absurd",
    "Abundant",   "Abusive",    "Acceptable", "Accessible", "Accidental", "Accurate",   "Acid",       "Acidic",     "Acoustic",   "Acrid",      "Actually",
    "Adhoc",      "Adamant",    "Adaptable",  "Addicted",   "Adhesive",   "Adjoining",  "Adorable",   "Afraid",     "Aggressive", "Agonizing",  "Agreeable",
    "Ahead",      "Ajar",       "Alcoholic",  "Alert",      "Alike",      "Alive",      "Alleged",    "Alluring",   "Aloof",      "Amazing",    "Ambiguous",
    "Ambitious",  "Amuck",      "Amused",     "Amusing",    "Ancient",    "Angry",      "Animated",   "Annoyed",    "Annoying",   "Anxious",    "Apathetic",
    "Aquatic",    "Aromatic",   "Arrogant",   "Ashamed",    "Aspiring",   "Assorted",   "Attractive", "Auspicious", "Automatic",  "Available",  "Average",
    "Awake",      "Aware",      "Awesome",    "Awful",      "Axiomatic",  "Bad",        "Barbarous",  "Bashful",    "Bawdy",      "Beautiful",  "Befitting",
    "Beneficial", "Bent",       "Berserk",    "Best",       "Better",     "Bewildered", "Big",        "Billowy",    "Bite-sized", "Bitter",     "Bizarre",
    "Black",      "Bloody",     "Blue",       "Blue-eyed",  "Blushing",   "Boiling",    "Boorish",    "Bored",      "Boring",     "Bouncy",     "Boundless",
    "Brainy",     "Brash",      "Brave",      "Brawny",     "Breakable",  "Breezy",     "Brief",      "Bright",     "Broad",      "Broken",     "Brown",
    "Bumpy",      "Burly",      "Bustling",   "Busy",       "Cagey",      "Callous",    "Calm",       "Capable",    "Capricious", "Careful",    "Careless",
    "Caring",     "Cautious",   "Ceaseless",  "Certain",    "Changeable", "Charming",   "Cheap",      "Cheerful",   "Chemical",   "Chief",      "Childlike",
    "Chilly",     "Chivalrous", "Chubby",     "Chunky",     "Clammy",     "Classy",     "Clean",      "Clear",      "Clever",     "Cloistered", "Cloudy",
    "Closed",     "Clumsy",     "Cluttered",  "Coherent",   "Cold",       "Colorful",   "Colossal",   "Combative",  "Common",     "Complete",   "Complex",
    "Concerned",  "Condemned",  "Confused",   "Conscious",  "Cooing",     "Cool",       "Courageous", "Cowardly",   "Crabby",     "Craven",     "Crazy",
    "Creepy",     "Crooked",    "Crowded",    "Cruel",      "Cuddly",     "Cultured",   "Cumbersome", "Curious",    "Curly",      "Curved",     "Curvy",
    "Cut",        "Cute",       "Cynical",    "Daffy",      "Daily",      "Damaged",    "Damaging",   "Damp",       "Dangerous",  "Dapper",     "Dark",
    "Dashing",    "Dazzling",   "Dead",       "Deadpan",    "Deafening",  "Dear",       "Debonair",   "Decisive",   "Decorous",   "Deep",       "Deeply",
    "Defeated",   "Defective",  "Defiant",    "Delicate",   "Delicious",  "Delightful", "Demonic",    "Delirious",  "Dependent",  "Depressed",  "Deranged",
    "Deserted",   "Detailed",   "Determined", "Devilish",   "Didactic",   "Different",  "Difficult",  "Diligent",   "Direful",    "Dirty",      "Disastrous",
    "Discreet",   "Disgusted",  "Disgusting", "Distinct",   "Disturbed",  "Divergent",  "Dizzy",      "Doubtful",   "Drab",       "Draconian",  "Dramatic",
    "Dreary",     "Drunk",      "Dry",        "Dull",       "Dusty",      "Dynamic",    "Eager",      "Early",      "Earthy",     "Easy",       "Eatable",
    "Economic",   "Educated",   "Efficient",  "Eight",      "Elastic",    "Elated",     "Elderly",    "Electric",   "Elegant",    "Elfin",      "Elite",
    "Eminent",    "Empty",      "Enchanted",  "Enchanting", "Endurable",  "Energetic",  "Enormous",   "Envious",    "Equable",    "Equal",      "Erect",
    "Erratic",    "Ethereal",   "Evanescent", "Evasive",    "Even",       "Excellent",  "Excited",    "Exciting",   "Exclusive",  "Exotic",     "Expensive",
    "Exuberant",  "Exultant",   "Fabulous",   "Faded",      "Faint",      "Fair",       "Faithful",   "Fallacious", "False",      "Familiar",   "Famous",
    "Fanatical",  "Fancy",      "Fantastic",  "Far",        "Far-flung",  "Fascinated", "Fast",       "Fat",        "Faulty",     "Fearful",    "Fearless",
    "Feeble",     "Feigned",    "Female",     "Fertile",    "Festive",    "Few",        "Fierce",     "Filthy",     "Fine",       "Finicky",    "First",
    "Five",       "Fixed",      "Flagrant",   "Flaky",      "Flashy",     "Flat",       "Flawless",   "Flimsy",     "Flippant",   "Flowery",    "Fluffy",
    "Fluttering", "Foamy",      "Foolish",    "Foregoing",  "Forgetful",  "Fortunate",  "Four",       "Frail",      "Fragile",    "Frantic",    "Free",
    "Freezing",   "Frequent",   "Fresh",      "Fretful",    "Friendly",   "Frightened", "Full",       "Fumbling",   "Functional", "Funny",      "Furry",
    "Furtive",    "Future",     "Futuristic", "Fuzzy",      "Gabby",      "Gainful",    "Gamy",       "Gaping",     "Garrulous",  "Gaudy",      "General",
    "Gentle",     "Giant",      "Giddy",      "Gifted",     "Gigantic",   "Glamorous",  "Gleaming",   "Glib",       "Glistening", "Glorious",   "Glossy",
    "Godly",      "Good",       "Goofy",      "Gorgeous",   "Graceful",   "Grandiose",  "Grateful",   "Gratis",     "Gray",       "Greasy",     "Great",
    "Greedy",     "Green",      "Grey",       "Grieving",   "Groovy",     "Grotesque",  "Grouchy",    "Grubby",     "Gruesome",   "Grumpy",     "Guarded",
    "Guiltless",  "Gullible",   "Gusty",      "Guttural",   "Habitual",   "Half",       "Hallowed",   "Halting",    "Handsome",   "Handsomely", "Handy",
    "Hanging",    "Hapless",    "Happy",      "Hard",       "Harmonious", "Harsh",      "Hateful",    "Heady",      "Healthy",    "Heavenly",   "Heavy",
    "Hellish",    "Helpful",    "Helpless",   "Hesitant",   "Hideous",    "High",       "Hilarious",  "Hissing",    "Historical", "Holistic",   "Hollow",
    "Homeless",   "Homely",     "Honorable",  "Horrible",   "Hospitable", "Hot",        "Huge",       "Hulking",    "Humdrum",    "Humorous",   "Hungry",
    "Hurried",    "Hurt",       "Hushed",     "Husky",      "Hypnotic",   "Hysterical", "Icky",       "Icy",        "Idiotic",    "Ignorant",   "Ill",
    "Illegal",    "Ill-fated",  "Imaginary",  "Immense",    "Imminent",   "Impartial",  "Imperfect",  "Impolite",   "Important",  "Imported",   "Impossible",
    "Incredible", "Infamous",   "Innate",     "Innocent",   "Insidious",  "Internal",   "Invincible", "Irate",      "Irritating", "Itchy",      "Jaded",
    "Jagged",     "Jazzy",      "Jealous",    "Jittery",    "Jobless",    "Jolly",      "Joyous",     "Judicious",  "Juicy",      "Jumbled",    "Jumpy",
    "Juvenile",   "Kaput",      "Keen",       "Kind",       "Kindly",     "Knotty",     "Knowing",    "Known",      "Labored",    "Lacking",    "Lame",
    "Lamentable", "Languid",    "Large",      "Last",       "Late",       "Laughable",  "Lavish",     "Lazy",       "Lean",       "Learned",    "Left",
    "Legal",      "Lethal",     "Level",      "Lewd",       "Light",      "Like",       "Likeable",   "Limping",    "Literate",   "Little",     "Lively",
    "Living",     "Lonely",     "Long",       "Longing",    "Long-term",  "Loose",      "Lopsided",   "Loud",       "Loutish",    "Lovely",     "Loving",
    "Low",        "Lowly",      "Lucky",      "Ludicrous",  "Lumpy",      "Lush",       "Luxuriant",  "Lying",      "Lyrical",    "Macabre",    "Macho",
    "Maddening",  "Madly",      "Magenta",    "Magical",    "Majestic",   "Makeshift",  "Male",       "Malicious",  "Mammoth",    "Maniacal",   "Many",
    "Marked",     "Massive",    "Married",    "Marvelous",  "Material",   "Mature",     "Mean",       "Measly",     "Meaty",      "Medical",    "Meek",
    "Mellow",     "Melodic",    "Melted",     "Merciful",   "Mere",       "Messy",      "Mighty",     "Military",   "Milky",      "Mindless",   "Miniature",
    "Minor",      "Miscreant",  "Misty",      "Mixed",      "Moaning",    "Modern",     "Moldy",      "Momentous",  "Motionless", "Muddled",    "Mundane",
    "Murky",      "Mushy",      "Mute",       "Mysterious", "Naive",      "Nappy",      "Narrow",     "Nasty",      "Natural",    "Naughty",    "Nauseating",
    "Near",       "Neat",       "Nebulous",   "Necessary",  "Needless",   "Needy",      "Neighborly", "Nervous",    "New",        "Next",       "Nice",
    "Nifty",      "Nimble",     "Nine",       "Nippy",      "Noiseless",  "Noisy",      "Nonchalant", "Nonstop",    "Normal",     "Nostalgic",  "Nosy",
    "Noxious",    "Null",       "Numberless", "Numerous",   "Nutritious", "Nutty",      "Oafish",     "Obedient",   "Obeisant",   "Obese",      "Obnoxious",
    "Obscene",    "Obsequious", "Observant",  "Obsolete",   "Obtainable", "Oceanic",    "Odd",        "Offbeat",    "Old",        "Omniscient", "One",
    "Onerous",    "Open",       "Opposite",   "Optimal",    "Orange",     "Ordinary",   "Organic",    "Ossified",   "Outgoing",   "Outrageous", "Oval",
    "Overjoyed",  "Overrated",  "Overt",      "Painful",    "Pale",       "Paltry",     "Panicky",    "Panoramic",  "Parallel",   "Parched",    "Past",
    "Pastoral",   "Pathetic",   "Peaceful",   "Penitent",   "Perfect",    "Periodic",   "Perpetual",  "Petite",     "Phobic",     "Physical",   "Picayune",
    "Pink",       "Piquant",    "Placid",     "Plain",      "Plant",      "Plastic",    "Plausible",  "Pleasant",   "Plucky",     "Pointless",  "Poised",
    "Polite",     "Political",  "Poor",       "Possessive", "Possible",   "Powerful",   "Precious",   "Premium",    "Present",    "Pretty",     "Previous",
    "Pricey",     "Prickly",    "Private",    "Probable",   "Productive", "Profuse",    "Protective", "Proud",      "Psychotic",  "Public",     "Puffy",
    "Pumped",     "Puny",       "Purple",     "Purring",    "Pushy",      "Puzzled",    "Puzzling",   "Quack",      "Quaint",     "Quick",      "Quickest",
    "Quiet",      "Quirky",     "Quixotic",   "Quizzical",  "Rabid",      "Racial",     "Ragged",     "Rainy",      "Rampant",    "Rapid",      "Rare",
    "Raspy",      "Ratty",      "Ready",      "Real",       "Rebel",      "Receptive",  "Recondite",  "Red",        "Redundant",  "Reflective", "Regular",
    "Relieved",   "Remarkable", "Repulsive",  "Resolute",   "Resonant",   "Rhetorical", "Rich",       "Right",      "Righteous",  "Rightful",   "Rigid",
    "Ripe",       "Ritzy",      "Roasted",    "Robust",     "Romantic",   "Roomy",      "Rotten",     "Rough",      "Round",      "Royal",      "Ruddy",
    "Rude",       "Rural",      "Rustic",     "Ruthless",   "Sable",      "Sad",        "Safe",       "Salty",      "Same",       "Sassy",      "Satisfying",
    "Savory",     "Scandalous", "Scarce",     "Scared",     "Scary",      "Scattered",  "Scientific", "Scrawny",    "Screeching", "Second",     "Secret",
    "Secretive",  "Sedate",     "Seemly",     "Selective",  "Selfish",    "Separate",   "Serious",    "Shaggy",     "Shaky",      "Shallow",    "Sharp",
    "Shiny",      "Shivering",  "Shocking",   "Short",      "Shrill",     "Shut",       "Shy",        "Sick",       "Silent",     "Silky",      "Silly",
    "Simple",     "Simplistic", "Sincere",    "Six",        "Skillful",   "Skinny",     "Sleepy",     "Slim",       "Slimy",      "Slippery",   "Sloppy",
    "Slow",       "Small",      "Smart",      "Smelly",     "Smiling",    "Smoggy",     "Smooth",     "Sneaky",     "Snobbish",   "Snotty",     "Soft",
    "Soggy",      "Solid",      "Somber",     "Sordid",     "Sore",       "Sour",       "Sparkling",  "Special",    "Spicy",      "Spiffy",     "Spiky",
    "Spiritual",  "Spiteful",   "Splendid",   "Spooky",     "Spotless",   "Spotted",    "Spotty",     "Spurious",   "Squalid",    "Square",     "Squealing",
    "Squeamish",  "Staking",    "Stale",      "Standing",   "Statuesque", "Steadfast",  "Steady",     "Stealthy",   "Steep",      "Sticky",     "Stiff",
    "Stingy",     "Stormy",     "Straight",   "Strange",    "Striped",    "Strong",     "Stupendous", "Stupid",     "Sturdy",     "Subdued",    "Subsequent",
    "Successful", "Succinct",   "Sudden",     "Sulky",      "Super",      "Superb",     "Supreme",    "Swanky",     "Sweet",      "Sweltering", "Swift",
    "Synonymous", "Taboo",      "Tacit",      "Tacky",      "Talented",   "Tall",       "Tame",       "Tan",        "Tangible",   "Tangy",      "Tart",
    "Tasteful",   "Tasteless",  "Tasty",      "Tawdry",     "Tearful",    "Tedious",    "Teeny",      "Teeny-tiny", "Telling",    "Temporary",  "Ten",
    "Tender",     "Tense",      "Tenuous",    "Terrible",   "Terrific",   "Tested",     "Testy",      "Thankful",   "Thick",      "Thin",       "Thinkable",
    "Third",      "Thirsty",    "Thoughtful", "Three",      "Thundering", "Tidy",       "Tight",      "Tiny",       "Tired",      "Tiresome",   "Toothsome",
    "Torpid",     "Tough",      "Towering",   "Tranquil",   "Trashy",     "Tremendous", "Tricky",     "Trite",      "Troubled",   "Truculent",  "True",
    "Truthful",   "Two",        "Typical",    "Ubiquitous", "Ugliest",    "Ugly",       "Ultra",      "Unable",     "Unadvised",  "Unarmed",    "Unbecoming",
    "Unbiased",   "Uncovered",  "Understood", "Unequal",    "Unequaled",  "Uneven",     "Unhealthy",  "Unique",     "Unkempt",    "Unknown",    "Unnatural",
    "Unruly",     "Unsightly",  "Unsuitable", "Untidy",     "Unused",     "Unusual",    "Unwieldy",   "Unwritten",  "Upbeat",     "Uppity",     "Upset",
    "Uptight",    "Used",       "Useful",     "Useless",    "Utopian",    "Utter",      "Uttermost",  "Vacuous",    "Vagabond",   "Vague",      "Valuable",
    "Various",    "Vast",       "Vengeful",   "Venomous",   "Verdant",    "Versed",     "Victorious", "Vigorous",   "Violent",    "Violet",     "Vivacious",
    "Voiceless",  "Volatile",   "Voracious",  "Vulgar",     "Wacky",      "Waggish",    "Waiting",    "Wakeful",    "Wandering",  "Wanting",    "Warlike",
    "Warm",       "Wary",       "Wasteful",   "Watery",     "Weak",       "Wealthy",    "Weary",      "Well-made",  "Well-off",   "Well-to-do", "Wet",
    "Whimsical",  "Whispering", "White",      "Whole",      "Wholesale",  "Wicked",     "Wide",       "Wide-eyed",  "Wiggly",     "Wild",       "Willing",
    "Windy",      "Wiry",       "Wise",       "Wistful",    "Witty",      "Woebegone",  "Womanly",    "Wonderful",  "Wooden",     "Woozy",      "Workable",
    "Worried",    "Worthless",  "Wrathful",   "Wretched",   "Wrong",      "Wry"
];

const nouns = [
    "Aardvark",   "Buffalo",    "Alligator",  "Ant",        "Anteater",   "Antelope",   "Ape",        "Armadillo",  "Donkey",      "Baboon",     "Badger",
    "Barracuda",  "Bat",        "Bear",       "Beaver",     "Bee",        "Bison",      "Boar",       "Bush",       "Butterfly",   "Camel",      "Calf",
    "Cat",        "Kitten",     "Cattle",     "Chamois",    "Cheetah",    "Chicken",    "Chick",      "Chimpanzee", "Infant",      "Empress",    "Troop",
    "Cobra",      "Cockroach",  "Cormorant",  "Cougar",     "Coyote",     "Crab",       "Crane",      "Crocodile",  "Crow",        "Deer",       "Dog",
    "Dogfish",    "Dolphin",    "Dove",       "Dragonfly",  "Duck",       "Dugong",     "Eagle",      "Eaglet",     "Echidna",     "Eel",        "Eland",
    "Elephant",   "Elk",        "Falcon",     "Ferret",     "Finch",      "Fly",        "Fox",        "Frog",       "Gaur",        "Gazelle",    "Gerbil",
    "Giant",      "Giraffe",    "Gnu",        "Goat",       "Goose",      "Gorilla",    "Guanaco",    "Guinea",     "Guineapig",   "Gull",       "Hamster",
    "Hare",       "Hawk",       "Hedgehog",   "Heron",      "Hornet",     "Horse",      "Human",      "Hyena",      "Iguana",      "Jackal",     "Jaguar",
    "Jellyfish",  "Kangaroo",   "Koala",      "Komodo",     "Kouprey",    "Kudu",       "Lark",       "Lemur",      "Leopard",     "Lion",       "Llama",
    "Loris",      "Louse",      "Lobster",    "Lyrebird",   "Magpie",     "Mallard",    "Manatee",    "Meerkat",    "Mink",        "Mole",       "Monkey",
    "Moose",      "Mouse",      "Mosquito",   "Mule",       "Okapi",      "Oryx",       "Ostrich",    "Otter",      "Owl",         "Ox",         "Oyster",
    "Panther",    "Partridge",  "Peafowl",    "Pelican",    "Penguin",    "Pig",        "Pigeon",     "Pony",       "Porcupine",   "Quelea",     "Rabbit",
    "Bunny",      "Raccoon",    "Rail",       "Ram",        "Rat",        "Raven",      "Reindeer",   "Rhino",      "Salamander",  "Sealion",    "Seal",
    "Seahorse",   "Seastar",    "Shark",      "Sheep",      "Shrew",      "Skunk",      "Snail",      "Snake",      "Spider",      "Squid",      "Squirrel",
    "Stinkbug",   "Swan",       "Tapir",      "Tarsier",    "Tiger",      "Toad",       "Turkey",     "Turtle",     "Vicuna",      "Walrus",     "Wasp",
    "Weasel",     "Whale",      "Wolf",       "Worm",       "Yak",        "Zebra",      "Hat",        "Cap",        "Beret",       "Astrakhan",  "Beanie",
    "Hardhat",    "Pillbox",    "Monkeycap",  "Operahat",   "Bonnet",     "Bowler",     "Coonskin",   "Fedora",     "Derby",       "Montero",    "Cowboyhat",
    "Sombrero",   "Yarmulke",   "Skullcap",   "Tam",        "Sunbonnet",  "Toque",      "Tophat",     "Babushka",   "Balaclava",   "Turban",     "Diadem",
    "Earmuffs",   "Visor",      "Scarf",      "Veil",       "Warbonnet",  "Pithhelmet", "Hood",       "Miter",      "Butter",      "Icecream",   "Cakebatter",
    "Coffee",     "Tea",        "Soda",       "Beer",       "Wine",       "Cappuccino", "Jell-o",     "Nougats",    "Lambchops",   "Steaks",     "Chowder",
    "Fishsoup",   "Spaghetti",  "Sushi",      "Fondue",     "Crabslegs",  "Shrimp",     "Onions",     "Bratwurst",  "Kielbasa",    "Hotdog",     "Hamburger",
    "Herbs",      "Grains",     "Legumes",    "Zampone",    "Casserole",  "Beans",      "Seeds",      "Stew",       "Cereal",      "Polenta",    "Pudding",
    "Pasta",      "Macaroni",   "Ravioli",    "Wafer",      "Crackers",   "Cookies",    "Sandwich",   "Gyro",       "Wrap",        "Omelet",     "Popcorn",
    "Walnuts",    "Nuts",       "Almonds",    "Pizza",      "Mousse",     "Brulee",     "Cakes",      "Pancake",    "Waffles",     "Toast",      "Candy",
    "Pie",        "Senator",    "Governor",   "Councilman", "Detective",  "Sleuth",     "Musician",   "Maestro",    "Conductor",   "Composer",   "Singer",
    "Architect",  "Physician",  "Manager",    "Usher",      "Painter",    "Model",      "Designer",   "Guest",      "Attorney",    "Lawyer",     "Judge",
    "Mayor",      "Therapist",  "Teacher",    "Principal",  "Professor",  "Orator",     "Man",        "Woman",      "Teen",        "Child",      "Mother",
    "Father",     "Sister",     "Brother",    "Uncle",      "Aunt",       "Son",        "Daughter",   "In-laws",    "Boy",         "Girl",       "Nurse",
    "Sibling",    "Settler",    "Pioneer",    "Waiter",     "Hostess",    "Host",       "Cashier",    "Attendant",  "Publisher",   "Witch",      "Warlock",
    "Ghost",      "Knight",     "Prince",     "Princess",   "Maiden",     "Godmother",  "Fairy",      "Petal",      "Sepal",       "Stamen",     "Pineboughs",
    "Bud",        "Branch",     "Blossom",    "Fruit",      "Bloom",      "Tree",       "Maple",      "Elm",        "Oak",         "Palm",       "Baobab",
    "Mangrove",   "Cyprus",     "Pine",       "Dogwood",    "Alder",      "Flowers",    "Rose",       "Tulip",      "Cyclamen",    "Lily",       "Carnations",
    "Wisteria",   "Flytrap",    "Hoe",        "Weeds",      "Plants",     "Canes",      "Palms",      "Apple",      "Lemon",       "Orange",     "Grapefruit",
    "Tangerine",  "Peach",      "Tomato",     "Banana",     "Vegetables", "Artichokes", "Leeks",      "Lettuce",    "Eggplants",   "Zucchini",   "Squash",
    "Pumpkin",    "Cabbage",    "Pepper",     "Onion",      "Garlic",     "Poison",     "Venom",      "Fire",       "Ship",        "Ferryboat",  "Oceanliner",
    "Oars",       "Sails",      "Dinghy",     "Yacht",      "Canoe",      "Catamaran",  "Gondola",    "Boat",       "Battleship",  "Clipper",    "Dhow",
    "Flatboat",   "Houseboat",  "Galleon",    "Frigate",    "Hydrofoil",  "Junk",       "Ketch",      "Yawl",       "Submarine",   "Schooner",   "Scow",
    "Flatbed",    "Suv",        "Van",        "Caboose",    "Train",      "Bullet",     "Metro",      "Subway",     "Cart",        "Taxi",       "Car",
    "Racingcar",  "Buggy",      "Dunebuggy",  "Dragster",   "Motorcycle", "Gokart",     "Limo",       "Stretch",    "Wagon",       "Trolley",    "Tram",
    "Bus",        "Parachute",  "Tractor",    "Trailer",    "Golfkart",   "Jeep",       "Bigrig",     "Bulldozer",  "Dumptruck",   "Towtruck",   "Engine",
    "Fireengine", "Policecar",  "Tank",       "Locomotive", "Ocean",      "Oasis",      "Sea",        "Lake",       "Saltlake",    "Seafoam",    "Waves",
    "Bubbles",    "Current",    "Waterbasin", "Bridge",     "Harbor",     "Pond",       "Wharf",      "Pier",       "Dock",        "Port",       "Shore",
    "Beach",      "Sandbar",    "Coast",      "River",      "Brook",      "Rivulet",    "Puddle",     "Waterfall",  "Cascades",    "Canal",      "Channel",
    "Stream",     "Creek",      "Marsh",      "Bog",        "Swamp",      "Bayou",      "Estuary",    "Whirlpool",  "Eddy",        "Geyser",     "Well",
    "Monsoon",    "Hurricane",  "Typhoon",    "Air",        "Snow",       "Rain",       "Sleet",      "Storm",      "Hail",        "Blizzard",   "Wind",
    "Breeze",     "Gale",       "Whirlwind",  "Maelstrom",  "Duststorm",  "Cloudburst", "Tornado",    "Twister",    "Clouds",      "Fog",        "Peasoup",
    "Floods",     "Flashflood", "Acidrain",   "Tremors",    "Lightning",  "Avalanche",  "Eclipse",    "Alpenglow",  "Tsunami",     "Waterspout", "Smog",
    "Aneroid",    "Barometer",  "Radiosonde", "Station",    "Map",        "Chalice",    "Bijou",      "Candelabra", "Menorah",     "Curio",      "Figurine",
    "Music-box",  "Objetd'art", "Trinket",    "Trims",      "Windchimes", "Birdcage",   "Birdbath",   "Cans",       "Urn",         "Bucket",     "Arrow",
    "Bow",        "Sword",      "Dart",       "Epee",       "Dagger",     "Hatchet",    "Pickax",     "Dolls",      "Broom",       "Mop",        "Pail",
    "Squeegee",   "Caddy",      "Telephone",  "Pipe",       "Paints",     "Brushes",    "Easel",      "Canvas",     "Trunk",       "Hook",       "Gun",
    "Glue",       "Tissue",     "Toilet",     "Kleenex",    "Papertowel", "Ropes",      "Rubber",     "Coil",       "Toys",        "Dogleash",   "Balloon",
    "Vases",      "Planters",   "Pen",        "Pad",        "Typewriter", "Computer",   "Laptop",     "Netbook",    "Stylus",      "Pencil",     "Desk",
    "Backpack",   "Shoerack",   "Notebook",   "Vellum",     "Chalk",      "Badge",      "Saddle",     "Spurs",      "Paper",       "Rollbook",   "Guestbook",
    "Pot",        "Plate",      "Dishes",     "Fork",       "Spoons",     "Knives",     "Knife",      "Samovar",    "Sky",         "Forest",     "Heaven",
    "Hell",       "Earth",      "Sun",        "Star",       "Planet",     "Mercury",    "Venus",      "Mars",       "Jupiter",     "Saturn",     "Uranus",
    "Neptune",    "Ceres",      "Pluto",      "Haumea",     "Makemake",   "Eris",       "Outerspace", "Town",       "Village",     "City",       "Country",
    "Farm",       "Suburb",     "Roads",      "Streets",    "Blocks",     "Zoo",        "Park",       "Museum",     "Cemetery",    "Tunnels",    "Caves",
    "Churches",   "Temples",    "Mall",       "Dresser",    "Armoire",    "Chiffonier", "Credenza",   "Console",    "Bookcase",    "Buffet",     "Armchair",
    "Recliner",   "Easychair",  "Bench",      "Banquette",  "Chair",      "Couch",      "Davenport",  "Sofa",       "Ottoman",     "Deckchair",  "Loveseat",
    "Highseat",   "Divan",      "Inglenook",  "Pew",        "Throne",     "Sectional",  "Stool",      "Pottychair", "Workbench",   "Nighttable", "Bed",
    "Daybed",     "Bassinet",   "Crib",       "Cradle",     "Cot",        "Futon",      "Hammock",    "Tatamimat",  "Waterbed",    "Trundlebed", "Hassock",
    "Hatrack",    "Stepstool",  "Footrest",   "Footstool",  "Tripod",     "Mirror",     "Nightlight", "Torchiere",  "Sunlamp",     "Spotlight",  "Ceilingfan",
    "Cupboard",   "Cardtable",  "Hutch",      "Locker",     "Wetbar",     "Vanity",     "Rack",       "Hopechest",  "Sculpture",   "Painting",   "Eye",
    "Pupil",      "Iris",       "Retina",     "Eyeball",    "Eyelids",    "Eyelashes",  "Eyebrows",   "Lap",        "Waist",       "Belly",      "Tummy",
    "Rearend",    "Crotch",     "Abdomen",    "Beard",      "Mustache",   "Sideburns",  "Fingernail", "Hand",       "Forearm",     "Arm",        "Knuckles",
    "Thumb",      "Wrist",      "Elbow",      "Leg",        "Toes",       "Knee",       "Ankle",      "Shin",       "Thigh",       "Hip",        "Breast",
    "Chest",      "Torso",      "Tongue",     "Lips",       "Gums",       "Mouth",      "Teeth",      "Bones",      "Spine",       "Throat",     "Lungs",
    "Kidneys",    "Intestines", "Colon",      "Spleen",     "Glands",     "Blood",      "Head",       "Skull",      "Brain",       "Muscle",     "Hair",
    "Xylophone",  "Clavier",    "Virginal",   "Lute",       "Drum",       "Frenchhorn", "Piano",      "Violin",     "Cello",       "Guitar",     "Flute",
    "Tuba",       "Harp",       "Mariachi",   "Orchestra",  "Oboe",       "Bassoon",    "Woodwinds",  "Brass",      "Viola",       "Kettledrum", "Peyotedrum",
    "Tambourine", "Tambour",    "Saxophone",  "Marimba",    "Maracas",    "Shofar",     "Cymbals",    "Kazoo",      "Dulcimer",    "Accordion",  "Lyre",
    "Fiddle",     "Banjo",      "Balalaika",  "Sitar",      "Ukulele",    "Zither",     "Bagpipes",   "Piccolo",    "Clarinet",    "Cornet",     "Panpipe",
    "Tuningfork", "Metronome",  "Castanets",  "Woofer",     "Sniper",     "Marksman",   "Cleaner",    "Pyro",       "Attacker",    "Mechanic",   "Janitor",
    "Scrubber",   "Garbageman", "Technician", "Ninja",      "Medic",      "Spy",        "Assassin",   "Gunman",     "Triggerman",  "Butcher",    "Killer",
    "Dodger",     "Booger",     "Engineer",   "Doctor",     "Surgeon",    "Fighter",    "Shooter",    "Gunner",     "Soldier",     "Officer",    "Veteran",
    "Scout",      "Mercenary",  "Commando",   "Cadet",      "Guard",      "Warrior",    "Trooper",    "Gambler",    "Specialist",  "Shaper",     "Finisher",
    "Gladiator",  "Boxer",      "Wrestler",   "Warlord",    "Rival",      "Armory",     "Agent",      "Rebel",      "Brawler",     "Bruiser",    "Bully",
    "Champion",   "Hero",       "Battler",    "Combatant",  "Fencer",     "Swordsman",  "Expert",     "Gangster",   "Gangsta",     "Bandit",     "Hoodlum",
    "Mobster",    "Robber",     "Thief",      "Burglar",    "Pirate",     "Thug",       "Hitman",     "Hitperson",  "Dealer",      "Desperado",  "Criminal",
    "Crook",      "Hijacker",   "Carjacker",  "Villain",    "Convict",    "Fugitive",   "Mug",        "Outlaw",     "Ruffian",     "Cutthroat",  "Devil",
    "Murderer",   "Psycho",     "Punk",       "ASBO",       "Offender",   "Drifter",    "Rioter",     "Goon",       "Roughneck",   "Brute",      "Hacker",
    "Cabbie",     "Wheeler",    "Driver",     "Rider",      "Cyclist",    "Cowboy",     "Operative",  "Carrier",    "Transporter", "Trucker",    "Wheelman",
    "Vampire",    "Parasite",   "Tramp",      "Bum",        "Hobo",       "Hitchhiker", "Deadbeat",   "Acrobat"
];

console.log("Adjectives:", adjectives.length);
console.log("Nouns:", nouns.length);
```

Commit:
Modify to remove trailing commas from arrays

Commit:
Modify the number of adjectives and nouns

Modifies the hardcoded numbers that determine the length of the arrays and leads to names containing a null value

TheNormalnij commented on Client/core/CNickGen.h:
```diff
@@ -9,8 +9,8 @@
-#define NICKGEN_NUM_ADJECTIVES      1048
-#define NICKGEN_NUM_NOUNS           934
+#define NICKGEN_NUM_ADJECTIVES      1040
+#define NICKGEN_NUM_NOUNS           921
```
you need to replace `const char* const CNickGen::m_szAdjectives` with `std::array<const char*, SIZE_HERE>` to avoid all possible problems

TheNormalnij replied:
so it will be a compile time error in the future

TracerDS commented on Client/core/CNickGen.cpp:
```diff
@@ -106,7 +106,7 @@ const char* const CNickGen::m_szAdjectives[] = {
     "Whimsical",  "Whispering", "White",      "Whole",      "Wholesale",  "Wicked",     "Wide",       "Wide-eyed",  "Wiggly",     "Wild",       "Willing",
     "Windy",      "Wiry",       "Wise",       "Wistful",    "Witty",      "Woebegone",  "Womanly",    "Wonderful",  "Wooden",     "Woozy",      "Workable",
-    "Worried",    "Worthless",  "Wrathful",   "Wretched",   "Wrong",      "Wry",
+    "Worried",    "Worthless",  "Wrathful",   "Wretched",   "Wrong",      "Wry"
```
why'd you remove the `,`?

Commit:
Requested changes

Commit:
Resolve errors

Resolve linker errors for CNickGen static members and add missing array include

TheNormalnij commented:
Closed in favor #3834
